pkg/resources/account_role.go

```diff
@@ -0,0 +1,253 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRoleSchema = map[string]*schema.Schema{
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"tag": tagReferenceSchema,
+}
+
+func AccountRole() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateAccountRole,
+		ReadContext:   ReadAccountRole,
+		DeleteContext: DeleteAccountRole,
+		UpdateContext: UpdateAccountRole,
+
+		Schema: accountRoleSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func CreateAccountRole(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
+	name := d.Get("name").(string)
+	id, err := helpers.DecodeSnowflakeParameterID(name)
```

Comment:
why are you doing a decode here?

Reply:
true changed to new acc object identifier